docs/BLUE_GREEN_MIGRATION.md

```diff
@@ -1,45 +1,86 @@
+
+Still figuring this out and testing
+
+## If a new DynamoDB table and Elasticsearch domain is necessary
+
+If a new dynamo table and elasticsearch domain is necessary, duplicate the modules found in `web-api/terraform/template/main.tf`. Be sure to update the version number at the end of the module names, `table_name`, and `domain_name`. Do not delete the old modules. Add the modules to `depends_on` in `main-east.tf` and `main-west.tf`
+
+1. Run a deploy in circle.
+2. Verify the new application works at: 
 	- https://<DEPLOYED_COLOR>-<ENV>.<ZONE_NAME>
 	- https://app-<DEPLOYED_COLOR>.<ENV>.<ZONE_NAME>
-6) Destroy the migration infrastructure to turn off the live streams
+3. Destroy the migration infrastructure to turn off the live streams
 	`DESTINATION_TABLE=b SOURCE_TABLE=a STREAM_ARN=abc npm run destroy:migration -- <DEPLOYED_ENV>`
 
 ## Errors You May Encounter
+
 ### AWS-related
+
 #### Route53 Error
+
 ```
 Error: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='_243f260ea635a6dffe0db2c6cc1c1158.*************************.', type='CNAME'] but it already exists]
 ```
 Manually delete the Route53 record and rerun the deploy.
 
 
 #### IAM Role already exists
+
 ```
 Error: Error creating IAM Role migration_role_<ENV>: EntityAlreadyExists: Role with name 		migration_role_<ENV> already exists.
 	status code: 409, request id: ***********
 
   on migration.tf line 1, in resource "aws_iam_role" "migration_role":
    1: resource "aws_iam_role" "migration_role" {
-   ```
-Delete the role in the AWS IAM console and rerun `npm run deploy:environment-specific <ENV>`.
+```
 
+AWS IAM console and rerun `npm run deploy:environment-specific <ENV>`.
 
 #### Failing smoke tests
-When this is run for the first time on a new environment, the smoke tests may fail for up to an hour after the initial deploy due to the header security lambda redeploying to all edge locations. To resolve, wait an hour and rerun the smoke tests.
 
+When this is run for the first time on a new environment, the smoke tests may fail for up to an hour after the initial deploy due to the header security lambda redeploying to all edge locations. To resolve, wait an hour and rerun the smoke tests.
 
 #### 403 on websockets endpoint
-If the websockets endpoint returns a 403 Unauthorized error (can be seen during trial session smoke tests), redeploy the websocket APIs for the environment in the AWS console under API Gateway (both east and west).
\ No newline at end of file
+
+If the websockets endpoint returns a 403 Unauthorized error (can be seen during trial session smoke tests), redeploy the websocket APIs for the environment in the AWS console under API Gateway (both east and west).
+
+#### Certificates
+
+```Error: Error creating API Gateway Domain Name: BadRequestException: The provided certificate does not exist.
+
+  on ../api/api-public.tf line 106, in resource "aws_api_gateway_domain_name" "api_public_custom":
```

Comment:
This looks to be an unfinished section

Reply:
You're right. It was more importing of existing resources, I think. Anyhow. Yeah, this is a work in progress as I'm going to continue to edit this doc as I get it going in all of the environments. Removed for now.